README.md

```diff
@@ -166,42 +167,75 @@ Alternatively you are able to use any `F[_]` which is convertible to/from `monix
+import com.avast.metrics.scalaapi.Monitor
+import com.typesafe.config.ConfigFactory
 import monix.eval._
+import monix.execution.Scheduler
+
+import scala.concurrent.duration.Duration
 
 val config = ConfigFactory.load().getConfig("myRabbitConfig")
 
-implicit val ec: ExecutionContext = ???
+implicit val sch: Scheduler = ???
+val monitor: Monitor = ???
+
 val blockingExecutor: ExecutorService = Executors.newCachedThreadPool()
 
-val monitor: Monitor = ???
+// the "FP" way:
+
+val consumerAndProducer: Task[(RabbitMQConsumer[Task], RabbitMQProducer[Task, Bytes])] = RabbitMQConnection
+  .fromConfig[Task](config, blockingExecutor)
+  .flatMap { connection =>
+    /*
+      Here you have created the connection; it's shared for all producers/consumers amongst one RabbitMQ server - they will share a single
+      TCP connection but have separated channels.
+      If you expect very high load, you can use separate connections for each producer/consumer, however it's usually not needed.
+    */
+
+    for {
+      consumer <- connection.newConsumer[Bytes]("consumer", monitor) {
+        case delivery: Delivery.Ok[Bytes] =>
+          println(delivery)
+          Task.now(DeliveryResult.Ack)
+
+        case _: Delivery.MalformedContent =>
+          Task.now(DeliveryResult.Reject)
+      }
 
-// here you create the connection; it's shared for all producers/consumers amongst one RabbitMQ server - they will share a single TCP connection
-// but have separated channels
-// if you expect very high load, you can use separate connections for each producer/consumer, but it's usually not needed
-val rabbitConnection = RabbitMQConnection.fromConfig[Task](config, blockingExecutor) // DefaultRabbitMQConnection[Task]
+      producer <- connection.newProducer("producer", monitor)
+    } yield {
+      (consumer, producer)
+    }
+  }
 
-val consumer = rabbitConnection.newConsumer[Bytes]("consumer", monitor) { 
+// the "common" way:
+
+val rabbitConnection = RabbitMQConnection.fromConfig[Task](config, blockingExecutor).runSyncUnsafe(Duration.Inf) // DefaultRabbitMQConnection[Task]
+
+val consumer = rabbitConnection.newConsumer[Bytes]("consumer", monitor) {
   case delivery: Delivery.Ok[Bytes] =>
-      println(delivery)
-      Task.now(DeliveryResult.Ack)
-      
-    case _: Delivery.MalformedContent =>
-      Task.now(DeliveryResult.Reject)
-} // DefaultRabbitMQConsumer[Task]
+    println(delivery)
+    Task.now(DeliveryResult.Ack)
 
-val sender = rabbitConnection.newProducer("producer", monitor) // DefaultRabbitMQProducer[Task]
+  case _: Delivery.MalformedContent =>
+    Task.now(DeliveryResult.Reject)
+}.runSyncUnsafe(Duration.Inf) // RabbitMQConsumer[Task]
 
-sender.send(...).runAsync // because it's Task, don't forget to run it ;-)
+val sender = rabbitConnection.newProducer("producer", monitor).runSyncUnsafe(Duration.Inf) // RabbitMQProducer[Task, Bytes]
 
+sender.send(...).runAsync // because it's Task, don't forget to run it ;-)
 ```
```

Comment:
By the way, you are doing a lot of examples and it would be good to be sure that they compile correctly. There is a tool called Tut that you could use: http://tpolecat.github.io/tut/

Reply:
I understand the problem you're addressing however:
1. It's plugin for SBT, this project use Gradle
1. I'd like to run it in "check" mode and it doesn't offer such thing, I'd need to hack it somehow.